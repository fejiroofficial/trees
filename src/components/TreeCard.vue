<template>
  <div class="tree-card ant-card ant-card-bordered">
    <div class="tree-card__name">{{ tree.name }}</div>
    <p>{{ tree.species_name }}</p>
    <img
      v-if="showImage"
      :alt="tree.name"
      :src="tree.image"
      style="width: 200px; height: 200px"
    />
    <button class="ant-btn" @click="showImage = !showImage" style="margin-top: 10px;">{{
      showImage ? "Hide Image" : "Show Image"
    }}</button>
  </div>
</template>

<script>
export default {
  props: {
    tree: Object,
  },
  data() {
    return {
      showImage: false,
    };
  },
};
</script>

<style lang="scss" scoped>
.tree-card {
  width: 300px;
  margin-bottom: 50px;
  padding: 20px;

  &__name {
    font-weight: bold;
    font-size: x-large;
  }
}
.ant-card {
  margin: 20px;
}
</style>
