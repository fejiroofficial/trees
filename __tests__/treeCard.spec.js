import { mount, createLocalVue } from "@vue/test-utils";
import TreeCard from "@/components/TreeCard";
import Card from "ant-design-vue/lib/card";
import Button from "ant-design-vue/lib/button";

const localVue = createLocalVue();
localVue.component(Button.name, Button);
localVue.component(Card.name, Card);

describe("TreeCard.vue", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(TreeCard, {
      localVue,
      propsData: {
        tree: {
          name: "sample tree",
          species_name: "sample species",
          image: "www.image.com",
        },
      },
    });
  });

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("displays props", () => {
    expect(wrapper.find(".tree-card__name").text()).toBe("sample tree");
    expect(wrapper.find("p").text()).toBe("sample species");
  });
});
