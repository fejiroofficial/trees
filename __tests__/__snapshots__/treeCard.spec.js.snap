// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TreeCard.vue renders correctly 1`] = `
<div
  class="tree-card ant-card ant-card-bordered tree-card__short"
>
  <div
    class="tree-card__name"
  >
    sample tree
  </div>
   
  <p>
    sample species
  </p>
   
  <div
    class="tree-card__flex-box"
  >
    <img
      alt="sample tree"
      class="tree-card__img"
      src="www.image.com"
      style="display: none;"
    />
  </div>
   
  <div
    class="tree-card__flex-box"
  >
    <button
      class="ant-btn tree-card__button"
    >
      
      Show Image
    
    </button>
  </div>
</div>
`;
